File: src/108. Convert Sorted Array to Binary Search Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */

/**
 * @param {number[]} nums
 * @return {TreeNode}
 */

export function sortedArrayToBST(nums) {
  return helper(0, nums.length)
  function helper(start, end) {
    if (start >= end) return null
    const mid = Math.floor((start + end) / 2)
    return new TreeNode(nums[mid], helper(start, mid), helper(mid + 1, end))
  }
}

File: src/108. Convert Sorted Array to Binary Search Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* sortedArrayToBST(vector<int>& nums) {
        return build(nums, 0, nums.size());
    }
private:
    TreeNode* build(vector<int>& nums, int start, int end) {
        if (start >= end) {
            return NULL;
        }
        int mid = (start + end) / 2;
        return new TreeNode(
            nums[mid],
            build(nums, start, mid),
            build(nums, mid + 1, end)
        );
    }
};